.github/workflows/validator.yml

```diff
@@ -0,0 +1,72 @@
+
+name: Dataset Validator
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  Validate:
+    runs-on: ubuntu-latest
+
+    steps:
+
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.7
+
+    #- name: Update software
+    #  run: |
+    # do we want to do this? it's helpful but also slow.
+    #    sudo apt-get update &&
+    #    sudo DEBIAN_FRONTEND=noninteractive apt-get -y dist-upgrade &&
+    #    sudo DEBIAN_FRONTEND=noninteractive apt-get autoremove
+
+    - name: Install git-annex
+      run: |
+        sudo apt-get install -y git-annex
+        git config --global annex.thin true
+
+    - name: Install bids-validator
+      run: |
+        # install proper NodeJS for bids-validator
+        curl -sL https://deb.nodesource.com/setup_current.x | sudo -E bash -
+        sudo apt-get install -y nodejs
+        sudo npm install -g bids-validator
+
+    - name: Install spine-generic for acquisition parameters check
+      run: pip install spinegeneric@git+https://github.com/spine-generic/spine-generic.git@master
+
+    #- name: Increase free space
+    #   run: |
+    #    # this takes about 2 minutes but saves about 30GB, which is space we might need for git-annex.
+    #    # annex.thin saves a lot of space, but if the dataset grows beyond what Github can handle
+    #    # try enabling this.
+    #    sudo rm -rf /usr/local/lib/android /usr/share/dotnet
+
+    - name: Checkout
+      uses: actions/checkout@v2
+
+    - name: git config
+      run: |
+        # this is needed by git-annex so that it can write to the local git-annex branch
+        # the tracking information about the local copy
+        git config --global user.name "gh-actions"
+        git config --global user.email "gh-actions"
+        # but actually, disable those writes. We don't need them because we're throwing away this copy immediately.
+        git config annex.alwayscommit false
+
+    - name: Download dataset
+      run: |
+        git fetch --depth=1 origin git-annex:git-annex # CI does a shallow checkout, so it is missing this important branch
```

Comment:
I think you could handle this in the `actions@checkout` step using [`ref:`](https://github.com/actions/checkout#usage) if you felt like it, but this seems like NBD.

Reply:
mmm I looked over that but I don't think so. The annoying thing (https://github.com/neuropoly/data-management/issues/67) is we need to checkout *two* branches, `git-annex` forces it upon us.

Reply:
This isn't actually checking out `git-annex`, it's just making sure the local copy in the background matches the remote copy.

Reply:
Ah wait, disregard this... we don't checkout this branch, we only want it to be available?

Reply:
Oh oops I commented before I refreshed the page, oops.